src/Illuminate/Foundation/Testing/CrawlerTrait.php

```diff
@@ -1027,15 +1027,25 @@ protected function assertFilterProducesResults($filter)
     /**
      * Filter elements according to the given name or ID attribute.
      *
-     * @param  string  $name
-     * @param  string  $element
+     * @param  string        $name
+     * @param  array|string  $elements
      * @return \Symfony\Component\DomCrawler\Crawler
      */
-    protected function filterByNameOrId($name, $element = '*')
+    protected function filterByNameOrId($name, $elements = '*')
     {
         $name = str_replace('#', '', $name);
 
-        return $this->crawler->filter("{$element}#{$name}, {$element}[name='{$name}']");
+        if (is_array($elements)) {
+            array_walk($elements, function (&$element) use ($name) {
+                $element = "{$element}#{$name}, {$element}[name='{$name}']";
+            });
+
+            $filter = implode(', ', $elements);
+        } else {
+            $filter = "{$elements}#{$name}, {$elements}[name='{$name}']";
+        }
+
+        return $this->crawler->filter($filter);
     }
 
```

Comment:
We could also do something like:

```php
protected function filterByNameOrId($name, $elements = '*')
{
    $name = str_replace('#', '', $name);

    if (! is_array($elements)) {
        $elements = preg_split('/,\s?|\|/', $elements);
    }

    array_walk($elements, function (&$element) use ($name) {
        $element = "{$element}#{$name}, {$element}[name='{$name}']";
    });

    return $this->crawler->filter(implode(', ', $elements));
}
```

then we could pass in any `$elements` like `"*"`, `"input, texarea"`, `"input|textarea"`, or `['input', 'textarea']` and it looks a little cleaner than what I have, and the tests still pass.